source/opt/iterator.h

```diff
@@ -81,6 +81,24 @@ class UptrVectorIterator
   inline typename std::enable_if<!IsConstForMethod, UptrVectorIterator>::type
   InsertBefore(Uptr value);
 
+  // Inserts the given |valueVector| to the position pointed to by this iterator
+  // and returns an iterator to the newly inserted |valueVector|.
+  // If the underlying vector changes capacity, all previous iterators will be
+  // invalidated. Otherwise, those previous iterators pointing to after the
+  // insertion point will be invalidated.
+  template <bool IsConstForMethod = IsConst>
+  inline typename std::enable_if<!IsConstForMethod, UptrVectorIterator>::type
+  MoveBefore(UptrVector& valueVector);
```

Comment:
Well, if the previous routine could be called InsertBefore, then I guess this one can be as well. They are generally doing the same thing.

I made valueVector a ref for efficiency, but will change it to a pointer.